src/data/ar-io-data-source.ts

```diff
@@ -136,6 +151,43 @@ export class ArIODataSource implements ContiguousDataSource {
     return this.peers[keys[randomIndex]];
   }
 
+  selectRandomWeightedPeers(count: number): string[] {
+    const log = this.log.child({ method: 'selectRandomWeightedPeers' });
+
+    if (this.weightedPeers.length === 0) {
+      log.warn('No weighted peers available');
+      throw new Error('No weighted peers available');
+    }
+
+    return randomWeightedChoices<string>({ table: this.weightedPeers, count });
+  }
+
+  handlePeerSuccess(peer: string): void {
+    metrics.getDataStreamSuccessesTotal.inc({
+      class: this.constructor.name,
+      peer,
+    });
+    // warm the succeeding peer
+    this.weightedPeers.forEach((weightedPeer) => {
+      if (weightedPeer.id === peer) {
+        weightedPeer.weight = Math.min(weightedPeer.weight + 0.1, 100);
```

Comment:
Let's make the warming and cooling values configurable.

Reply:
I introduced env-var WEIGHTED_PEERS_TEMPERATURE_DELTA